service/src/main/kotlin/io/provenance/explorer/service/AssetService.kt

```diff
@@ -124,6 +129,48 @@ class AssetService(
         PagedResults(res.pagination.total.pageCountOfResults(count), list, res.pagination.total)
     }
 
+    fun getTokenDistributionStats() = transaction { TokenDistributionAmountsRecord.getStats() }
+
+    fun updateTokenDistributionStats(denom: String) = accountService.getCurrentSupply(denom).let { supply ->
+        val res = markerClient.getAllMarkerHolders(denom)
+        // TODO: Need to increase the request timeout to prevent a
+        //       DEADLINE_EXCEEDED: deadline exceeded after 9.999915266s. [remote_addr=localhost/127.0.0.1:9090]
+
+        val assetHolders = res.balancesList.map { bal ->
+            val balance = bal.coinsList.first { coin -> coin.denom == denom }.amount
+            AssetHolder(bal.address, CountStrTotal(balance, supply, denom))
+        }.sortedByDescending { it.balance.count.toBigInteger() }
+
+        val size = assetHolders.size
+        val totalSupply = assetHolders[0].balance.total?.toBigDecimal()!!
```

Comment:
This would be the sum total. Or, you could just use the marker grpc func for total supply. Seems less prone to error.